client/a8c-for-agencies/data/marketplace/use-products-query.ts

```diff
@@ -0,0 +1,99 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import selectAlphabeticallySortedProductOptions from 'calypso/jetpack-cloud/sections/partner-portal/lib/select-alphabetically-sorted-product-options';
+import wpcom from 'calypso/lib/wp';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getActiveAgencyId } from 'calypso/state/a8c-for-agencies/agency/selectors';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { APIProductFamily, APIProductFamilyProduct } from 'calypso/state/partner-portal/types';
+
+function queryProducts(
+	isPublicFacing: boolean,
+	agencyId?: number
+): Promise< APIProductFamily[] > {
+	const productsAPIPath = isPublicFacing
+		? '/jetpack-licensing/public/manage-pricing'
+		: '/jetpack-licensing/partner/product-families';
+
+	return wpcom.req
+		.get(
+			{
+				apiNamespace: 'wpcom/v2',
+				path: productsAPIPath,
+			},
+			{
+				...( agencyId && { agency_id: agencyId } ),
+			}
+		)
+		.then( ( data: APIProductFamily[] ) => {
+			const exclude = [
+				'free',
+				'personal',
+				'premium',
+				'professional',
+				'jetpack-backup-daily',
+				'jetpack-backup-realtime',
+				'jetpack-backup-t0',
+				'jetpack-security-daily',
+				'jetpack-security-realtime',
+			];
+
+			return data
+				.map( ( family ) => {
+					return {
+						...family,
+						products: family.products
+							.filter( ( product ) => {
+								return exclude.indexOf( product.slug ) === -1;
+							} )
+							.map( ( product ) => ( {
+								...product,
+								family_slug: family.slug,
+							} ) ),
+					};
+				} )
+				.filter( ( family ) => {
+					return family.products.length > 0;
+				} );
+		} );
+}
+
+export function usePublicProductsQuery(): UseQueryResult< APIProductFamilyProduct[], unknown > {
+	return useProductsQuery( true );
+}
+
+export default function useProductsQuery(
+	isPublicFacing = false
```

Comment:
We will probably not need this flag in A4A. I think we can safely remove this one.

Reply:
I saw some places where we are using it. So, I decided to keep it for now.